crypto/s2n_fips.c

```diff
@@ -17,16 +17,35 @@
 
 #include "crypto/s2n_fips.h"
 
-/* Return 1 if FIPS mode is enabled, 0 otherwise. */
-int s2n_is_in_fips_mode()
+static int s2n_fips_mode = 0;
+
+int s2n_fips_init(void)
 {
 #ifdef OPENSSL_FIPS
     /* FIPS mode can be entered only if OPENSSL_FIPS is defined */
     if (FIPS_mode()) {
-        return 1;
+        s2n_fips_mode = 1;
+    } else {
+        s2n_fips_mode = 0;
     }
 #endif
 
-    /* FIPS mode is not supported or FIPS mode is not enabled */
     return 0;
 }
+
+int s2n_fips_cleanup(void)
+{
+    s2n_fips_mode = 0;
```

Comment:
Originally, the intention was to ensure that s2n_init was always initialized with s2n_fips_mode equal to 0. I have removed s2n_fips_cleanup and have modified s2n_fips_init to always set s2n_fips_mode to 0 if FIPS_mode is not set.